client/src/main/java/org/spine3/client/CommandFactory.java

```diff
@@ -20,17 +20,39 @@
 
 package org.spine3.client;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Any;
 import com.google.protobuf.Message;
+import org.spine3.annotation.Internal;
+import org.spine3.base.ActorContext;
 import org.spine3.base.Command;
 import org.spine3.base.CommandContext;
 import org.spine3.base.Commands;
+import org.spine3.protobuf.AnyPacker;
+import org.spine3.time.ZoneOffset;
+import org.spine3.users.TenantId;
+import org.spine3.users.UserId;
+import org.spine3.validate.ConstraintViolationThrowable;
+
+import javax.annotation.Nullable;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spine3.time.Time.getCurrentTime;
+import static org.spine3.validate.Validate.checkValid;
 
 /**
  * Public API for creating {@link Command} instances, using the {@code ActorRequestFactory}
  * configuration.
  *
+ * <p>During the creation of {@code Command} instances the source {@code Message} instances, passed
+ * into creation methods, are validated. The validation is performed according to the constraints
+ * set in Protobuf definition of each {@code Message}. In case the message isn't valid,
+ * an {@linkplain ConstraintViolationThrowable exception} is thrown.
+ *
+ * <p>Therefore it is recommended to use a corresponding
+ * {@linkplain org.spine3.validate.ValidatingBuilder validating Builder} implementation to create
```

Comment:
Why not `ValidatingBuilder`? After all we refer to particular kind of builders that we generate.